src/lib.rs

```diff
@@ -154,18 +162,24 @@ pub fn run(
     statsd: StatsdClient,
     workers: usize,
 ) -> Result<Server, std::io::Error> {
+    // Curious why you needed to use a Once synchronization for initialization
     START.call_once(|| {
         magick_wand_genesis();
     });
 
     let configuration = Data::new(configuration);
 
+    // I don't understand why moving this value into the closure without being
+    // clonable is a problem since you have no intent to use this reference
+    // again. It is correct, but I don't understand why.
+    // Did you ever learn why you had to use an Arc here? I see you keep moving the statsd value.
+    // I wonder if this API is only intended for values that could be taken ownership of? If so,
+    // maybe the intent is to provide a client like statsd through that interface? Like app_data?
```

Comment:
I was following this part of the [Cadence](https://github.com/56quarters/cadence) crate documentation which seemed to say that an `Arc` was required to share the StatsD client between threads: https://docs.rs/cadence/0.28.0/cadence/struct.StatsdClient.html#wrapping-with-an-arc

but maybe I misunderstood the documentation

Reply:
The docs are correct. Its just an awkward API to have to keep threading the statsd client by moving and cloning within each closure. I think it might be because the HttpServer type actually starts multiple Workers which each get their own Application.